pkg/drivers/generic/generic.go

```diff
@@ -25,6 +26,40 @@ const (
 	defaultMaxIdleConns = 2 // copied from database/sql
 )
 
+type Config struct {
+	MetricsRegisterer    prometheus.Registerer
+	Endpoint             string
+	Scheme               string
+	DataSourceName       string
+	ConnectionPoolConfig ConnectionPoolConfig
+	BackendTLSConfig     tls.Config
+}
+
+type Constructor func(ctx context.Context, cfg *Config) (leaderElect bool, backend server.Backend, err error)
+
+var driverRegistry map[string]Constructor = map[string]Constructor{}
+var defaultDriver string
+
+func RegisterDriver(driver string, constructor Constructor) {
+	driverRegistry[driver] = constructor
+}
+
+func SetDefaultDriver(driver string) {
```

Comment:
Please add doc comments as well, since this is going to be an explicit public extension point.

```suggestion
func RegisterDriver(scheme string, constructor Constructor) {
	driverRegistry[scheme] = constructor
}

func SetDefaultDriver(scheme string) {
```